Move kernel/soc/arch/drivers to new logger

nashif opened the pull request:
~~Still pending fix for #10026~~

Olivier-ProGlove committed:
drivers: watchdog: Migrate to new logging subsys

Migrate from `SYS_LOG` to `LOG` logging mechanism.

Signed-off-by: Olivier Martin <olivier.martin@proglove.de>
Signed-off-by: Anas Nashif <anas.nashif@intel.com>

Olivier-ProGlove committed:
subsys: dfu: Migrate to new logging subsys

Migrate from `SYS_LOG` to `LOG` logging mechanism.

Signed-off-by: Olivier Martin <olivier.martin@proglove.de>

Olivier-ProGlove committed:
drivers: gpio: Migrate to new logging subsys

Migrate from `SYS_LOG` to `LOG` logging mechanism.

Signed-off-by: Olivier Martin <olivier.martin@proglove.de>

nashif committed:
kernel: move to new logger

Use the new logger framework for kernel.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
arch: setup logging using new logger

Use the new logger framework for architecture code.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
soc: logging: move soc tree to new logger

Use the new logger framework for soc related code.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif commented:
this obsoletes #9358 #9361 #9359

nashif committed:
kernel: move malloc handling to new logger

Make this part of the kernel log domain, it is used by the kernel.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
drivers: entropy: remove unused logger kconfig

This is not being used by any drivers, so removing until we need it.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif commented:
also obsoletes #9428 and #9425

nashif committed:
drivers: adc: move to new logger

Move to new logger subsystem.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

MaureenHelm commented on kernel/userspace.c:
```diff
@@ -221,7 +225,7 @@ void *_impl_k_object_alloc(enum k_objects otype)
 	dyn_obj = z_thread_malloc(sizeof(*dyn_obj) + obj_size_get(otype));
 	if (!dyn_obj) {
-		SYS_LOG_WRN("could not allocate kernel object");
+		LOG_WRN("could not allocate kernel object");
```
This the only place we use logging in the kernel??

nashif replied:
right now yes, I think we will have more once this is moved to the new logger.

nashif committed:
drivers: audio: move to new logger

Move to new logger subsystem.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
drivers: can: move to new logger

Move to new logger subsystem.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
drivers: clock_control: move to new logger

Move to new logger subsystem.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
drivers: crypto: move to new logger

Move to new logger subsystem.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif commented:
obsoletes #9429 as well

nashif committed:
drivers: i2c: move to new logger

Move to new logger subsystem.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
drivers: dma: move to new logger

Move to new logger and adapt samples.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif commented:
> @nashif : isn't the new logger a bit immature still?

no, afaik it is ready to go and we need to move everyone to the new logger in 1.14.

nordic-krch commented on arch/arm/core/cortex_m/mpu/arm_mpu_v7_internal.h:
```diff
@@ -8,6 +8,11 @@
+
+#define LOG_LEVEL CONFIG_ARCH_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_DECLARE(mpu);
```
I think that even when #10045 is introduced this will not work. In arm_mpu.c (which includes this file) there will be 2 implementations of __log_current_const_data_get. We need to come up with a solution for logging in headers.

pizi-nordic replied:
Unfortunately, you cannot declare/register module in header file. It will end in conflict when you try to include such file in C file which declares logger module too.

This applies also to other subsystems.

Olivier-ProGlove commented:
@nashif rather than duplicate my work (my pull-requests were clearly marked with the appropriate tag - see: https://github.com/zephyrproject-rtos/zephyr/labels/area%3A%20Logging) you could have cherry-picked my patches. There was not much work left as I was waiting for the v1.13 release and the pull-requests [`logging: add template for log configuration`](https://github.com/zephyrproject-rtos/zephyr/pull/9933) and [`log: make name param explicit`](https://github.com/zephyrproject-rtos/zephyr/pull/9450) to be merged to clean my pull-requests.

You might actually have cherry-picked my patches but remove my credit (see https://github.com/zephyrproject-rtos/zephyr/pull/9429#issuecomment-422175587)...
If you are not aware there is actually a command named [`git cherry-pick`](https://git-scm.com/docs/git-cherry-pick) for that purpose.

I did this boring work to help the Open Source project to have the logging subsystem to be merged into v1.13 as [it was initially planned](https://github.com/zephyrproject-rtos/zephyr/issues/6186#ref-issue-331299022) and also to not need [my other pull-request that I think is needed in v1.13 without the new logging subsystem](https://github.com/zephyrproject-rtos/zephyr/pull/9046).

pizi-nordic commented on arch/arm/core/cortex_m/mpu/arm_core_mpu.c:
```diff
@@ -10,7 +10,10 @@
 #include <arch/arm/cortex_m/mpu/arm_core_mpu.h>
-#include <logging/sys_log.h>
+
+#define LOG_LEVEL CONFIG_ARCH_LOG_LEVEL
```
IMHO the MPU-related code should have separate configuration of log level.

pizi-nordic commented on arch/arc/core/mpu/arc_core_mpu.c:
```diff
@@ -9,7 +9,10 @@
 #include <arch/arc/v2/mpu/arc_core_mpu.h>
-#include <logging/sys_log.h>
+
+#define LOG_LEVEL CONFIG_ARCH_LOG_LEVEL
```
In my opinion we should have separate log level configuration for MPU (as it is separate subsystem).

pizi-nordic commented on arch/arm/core/cortex_m/mpu/arm_mpu_v8_internal.h:
```diff
@@ -9,6 +9,9 @@
 #include <cmse.h>
+#define LOG_LEVEL CONFIG_ARCH_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_DECLARE(mpu);
```
Unfortunately, you cannot declare/register module in header file. It will end in conflict when you try to include such file in C file which declares logger module too.

nashif commented:
@Olivier-ProGlove 

> @nashif rather than duplicate my work

I was not duplicating your work, the changes I have done were inclusive and covered everything in the tree, not selected architectures or drivers. Many of you PRs were stale and needed attention and I did not want to cherrypick something that was not ready and that could cause confusion on my part due to the old state they are in.

- I used the new API for LOG_MODULE_DECLARE
- I used the Kconfig template
- I covered ALL architectures and files
- I applied a mew generic domain scheme
- ...

I apologise if that caused any discomfort or confusion, that was not the intention.
Thank you for you contributions and we value your input.

pizi-nordic commented on arch/arm/core/cortex_m/mpu/nxp_mpu.c:
```diff
@@ -10,10 +10,13 @@
 
+#define LOG_LEVEL CONFIG_ARCH_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_DECLARE(mpu);
```
Another incarnation of problem mentioned earlier: The ARM driver registers the module and other MPU-related files uses this registration. However as ARM MPU driver is not used on NXP platforms, here you are trying to use unregistered module.

pizi-nordic commented on arch/arm/core/cortex_m/mpu/nxp_mpu.c:
```diff
@@ -491,7 +494,7 @@ static int nxp_mpu_init(struct device *arg)
 }
 
-#if defined(CONFIG_SYS_LOG)
+#if defined(CONFIG_LOG)
```
I am not sure is this is still valid.
@nordic-krch: As far as I remember logger could accept messages very early, before kernel initialization. Is that still true?

pizi-nordic commented on drivers/adc/adc_dw.c:
```diff
@@ -19,7 +19,6 @@
 #define ADC_CONTEXT_USES_KERNEL_TIMER
 #include "adc_context.h"
 #include "adc_dw.h"
-#include <logging/sys_log.h>
```
There is no log module registration/declaration in this file. Is that correct?

pizi-nordic commented on drivers/crypto/crypto_ataes132a.c:
```diff
@@ -24,6 +21,10 @@
 
+#define LOG_LEVEL CONFIG_CRYPTO_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ataes132a);
```
Here you are registering a driver-specific logging module, however in other subsystems all drivers had the same name in logger (for example "adc").

IMHO we should register every driver using it's name (like it is done here) as this allows for fine grained log control over console as well as does not produce conflicts if 2 drivers form same subsystem are simultaneously compiled-in.

nashif replied:
most systems would have only one peripheral from each type so the use of the subsystem here is cleaner IMO, but I am fine changing this and did for most of the drivers. Some become very long, like 'clock_control_xxxxx_xxxxx', IMO this is not nice in a log...

nashif committed:
samples: fix prj.conf to use new logger

Configure samples to use new logger.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: crypto: move to new logger

Move sample to use new logger.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: move crypto sample to new logger

move sample to use new logger.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
driver: pwm: use new logger

move PWM driver to use new logger.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: led_lp5562: move to new logger

move sample to use new logger.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: led_lp3943: move to new logger

move sample to use new logger.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: led_ws2812: move to new logger

move sample to use new logger.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: led_pca9633: move to new logger

move sample to use new logger.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: led_lpd8806: move to new logger

move sample to use new logger.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: flash_shell: move to new logger

move sample to use new logger.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
driver: spi: use new logger

move SPI drivers to use new logger.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: crypto: use printk instead of logging

Remove single use of logger in favor of printk.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
arch: nxp_mpu: fix types in log macro

Fix format type in log message.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
drivers: spi_dw: remove debug message

Remove debug message using undefined functions.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif commented:
@nordic-krch any idea why samples/drivers/crypto is failing here?

nashif committed:
drivers: display: define DISPLAY log level

Use a common log level for display drivers.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: logging: disable default log modules

log modules of kernel are creating noise, set them to off to have this
test work as expected.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: display: set log level correctly

Use boolean kconfig for setting log level.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
subsys: nvs: move to new logger

Move to new logger and update all related samples and configs.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
nvs: fix style

Fix code style after moving to new logger.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif commented:
@jarz-nordic @nordic-krch @Olivier-ProGlove review please

nashif committed:
gpio: log: rename level variable

Rename log level variable:
 CONFIG_LOG_GPIO_LEVEL to CONFIG_GPIO_LOG_LEVEL

Use template for log levels.

Register modules in drivers using logging.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
dfu: log: rename level variable

Rename log level variable:
CONFIG_LOG_IMG_MANAGER_LEVEL to CONFIG_IMG_MANAGER_LOG_LEVEL

Use template for log levels.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
gpio_sch: fix log message

log message missing a parameter for the pin.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
console: move to new logger

Use new logger and remove usage of SYS_LOG.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
telnet_console: fix syntax

Fix layout and syntax with uncrustify after changing to new logger.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
websocket_console: fix syntax

Fix layout and syntax with uncrustify after changing to new logger.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: led: fix default log level kconfig

Use correct Kconfig for default log level.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: crypto: use CONFIG_LOG

Use CONFIG_LOG instead of old CONFIG_SYS_LOG. The drivers in questions
already moved to new logger.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
ztest: remove unused include of sys_log.h

Remove include that is not being used in the file.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
